Add new unit tests and some refactoring to make testing easier

extramileit opened the pull request:
- Extract TransactionRequestDtoFactory from CardanoTokenSender
- Extract EligibleDonationsService from DonationJob
- Use @Inject instead of constructor dependencies in various places
- Add BlockChainServicesFactory to build all service beans related to BlockFrost API

extramileit committed:
Add new unit tests and some refactoring to make testing easier
- Extract TransactionRequestDtoFactory from CardanoTokenSender
- Extract EligibleDonationsService from DonationJob
- Use @Inject instead of constructor dependencies in various places
- Add BlockChainServicesFactory to build all service beans related to BlockFrost API

extramileit commented:
Hi Matti
Please feel free to ignore this PR, if you like.  Your original code is well designed and high quality!  The refactoring you see in the PR has mainly to do with making unit testing easier.  My PR ended up being far more invasive than I wanted.  And you won't hurt my feelings by discarding the PR.   I didn't do a comprehensive job writing tests either.  I only wrote a couple, so that I could spend time with the code and learn how to the blockfrost/bloxbean API.

On the other hand, you may find some value in the refactoring, which would be great. I'd also be glad to collaborate with you on other Java project needs.  I've been trying to learn smart contracts using Haskell and more recently trying to use the PAB beta codebase.  That has been rather frustrating, and I feel like I'm not accomplishing much.  Perhaps, I should continue to keep a hand in the Java work, as it appears one can accomplish a lot using Blockfrost/Bloxbean APIs.  Are you working on Bloxbean API, btw?

Thanks
Kurt

Cardano-Fans commented:
Kurt... I will have a look, please join Cardano Fans on telegram: https://t.me/cardano_fans

Also reminder, on Fridays, 13.15 - 14.15 UTC we have special ZOOM / Google Meet session for developers.

extramileit commented:
Hi, thanks for the invite to your Telegram group and the developer meeting.  I added myself to the cardano_fans telegram group.  Unfortunately, I wasn't paying close attention to the timezone of the DEV meeting, so I missed that.  I should be able to make the next one, though. :)
Thanks

andreAmorimF commented on src/main/java/crfa/app/infrastructure/DonationAccountFactory.java:
```diff
@@ -0,0 +1,53 @@
+        } catch (IOException e) {
+            var msg = "Unable to read: " + walletPassFile;
+            log.error(msg, e);
+            System.exit(-1);
```
do the exception will be thrown after the call on `System.exit` ?

extramileit replied:
Nice catch. I got rid of the System.exit and just rely on the bean instantiation to fail, which prevents application from starting up.

andreAmorimF commented on src/main/java/crfa/app/service/transaction/TransactionRequestDto.java:
```diff
@@ -0,0 +1,23 @@
+        this.transactionDetailsParams = transactionDetailsParams;
+        this.metadata = metadata;
+    }
+}
```
add newline

extramileit replied:
done

andreAmorimF commented on build.gradle:
```diff
@@ -65,3 +71,7 @@ java {
+
+test {
+    useJUnitPlatform()
+}
```
add newline

extramileit replied:
done

andreAmorimF commented on src/main/java/crfa/app/service/transaction/TransactionRequestDtoFactory.java:
```diff
@@ -0,0 +1,76 @@
+                .put(new BigInteger(CIP20_META_DATUM_LABEL), memo);
+    }
+
+}
```
add newline

andreAmorimF commented on src/test/java/crfa/app/service/CardanoTokenSenderTest.java:
```diff
@@ -0,0 +1,91 @@
+        return new Donation("1", 1, 1, null, Cadence.EPOCH, "entity", address, amount, null);
+    }
+
+}
```
add newline

extramileit replied:
done

andreAmorimF commented on src/test/java/crfa/app/service/CardanoTokenSenderTest.java:
```diff
@@ -0,0 +1,91 @@
+
+@SuppressWarnings("unchecked")
+@ExtendWith(MockitoExtension.class)
+class CardanoTokenSenderTest {
```
👍🏼

extramileit committed:
Tweaks based on review of PR#11
- In DonationAccountFactory, instead of System.exit rely on Bean instantiation failure to prevent startup, when wallet pass file cannot be read
- Small tweak to DonationJob to prevent storing the donation transactions in case transaction send did not throw exception but was not successful
- Add newline to various source files

extramileit committed:
Merge pull request #1 from woofpool/kurt/fix-pr#11

Tweaks based on review of PR#11

extramileit committed:
Additional fix for newline at end of file

extramileit committed:
Merge remote-tracking branch 'origin/main'

extramileit commented:
Thanks for the review.  Fixed changes.

matiwinnetou commented:
Thank You very much!
